chains/orchestrator-relays/runtime/starlight/src/lib.rs

```diff
@@ -0,0 +1,3813 @@
+    pallet_balances::Call as BalancesCall,
+    primitives::{AccountId, Balance},
+};
+
+#[cfg(feature = "runtime-benchmarks")]
+use snowbridge_core::{AgentId, TokenId};
+
+/// Constant values used within the runtime.
+use starlight_runtime_constants::{currency::*, fee::*, snowbridge::EthereumLocation, time::*};
+
+// XCM configurations.
+pub mod xcm_config;
+
+pub mod bridge_to_ethereum_config;
+
+// Weights
+mod weights;
+
+// Governance and configurations.
+pub mod governance;
+use {
+    governance::{
+        pallet_custom_origins, AuctionAdmin, Fellows, GeneralAdmin, Treasurer, TreasurySpender,
+    },
+    pallet_collator_assignment::CoreAllocationConfiguration,
+    xcm_runtime_apis::fees::Error as XcmPaymentApiError,
+};
+
+#[cfg(test)]
+mod tests;
+
+pub mod genesis_config_presets;
+
+impl_runtime_weights!(starlight_runtime_constants);
+
+// Make the WASM binary available.
+#[cfg(feature = "std")]
+include!(concat!(env!("OUT_DIR"), "/wasm_binary.rs"));
+
+/// Runtime version (Starlight).
+#[sp_version::runtime_version]
+pub const VERSION: RuntimeVersion = RuntimeVersion {
+    spec_name: Cow::Borrowed("starlight"),
+    impl_name: Cow::Borrowed("tanssi-starlight-v2.0"),
+    authoring_version: 0,
+    spec_version: 1300,
+    impl_version: 0,
+    apis: RUNTIME_API_VERSIONS,
+    transaction_version: 26,
+    system_version: 1,
+};
+
+/// The BABE epoch configuration at genesis.
+pub const BABE_GENESIS_EPOCH_CONFIG: babe_primitives::BabeEpochConfiguration =
+    babe_primitives::BabeEpochConfiguration {
+        c: PRIMARY_PROBABILITY,
+        allowed_slots: babe_primitives::AllowedSlots::PrimaryAndSecondaryVRFSlots,
+    };
+
+/// Native version.
+#[cfg(any(feature = "std", test))]
+pub fn native_version() -> NativeVersion {
+    NativeVersion {
+        runtime_version: VERSION,
+        can_author_with: Default::default(),
+    }
+}
+
+/// Aggregate message origin for the `MessageQueue` pallet.
+///
+/// Can be extended to serve further use-cases besides just UMP. Is stored in storage, so any change
+/// to existing values will require a migration.
+#[derive(Encode, Decode, Clone, MaxEncodedLen, Eq, PartialEq, RuntimeDebug, TypeInfo)]
+pub enum AggregateMessageOrigin {
+    /// Inbound upward message.
+    #[codec(index = 0)]
+    Ump(UmpQueueId),
+
+    /// The message came from a snowbridge channel. It will be processed by `snowbridge_pallet_outbound_queue`.
+    #[codec(index = 1)]
+    Snowbridge(ChannelId),
+
+    /// The message came from a snowbridge channel, and it's a custom message that only exists in Tanssi.
+    /// This will be processed by `CustomProcessSnowbridgeMessage`.
+    #[codec(index = 2)]
+    SnowbridgeTanssi(ChannelId),
+}
+
+#[cfg(feature = "runtime-benchmarks")]
+impl From<u32> for AggregateMessageOrigin {
+    fn from(n: u32) -> Self {
+        // Some dummy for the benchmarks.
+        Self::Ump(UmpQueueId::Para(n.into()))
+    }
+}
+
+pub struct GetAggregateMessageOrigin;
```

Comment:
perhaps we should start thinking about putting this in primitives, as they are analogous in both runtimes

Reply:
sure, let's plan for the next PR.